host/sgxload.c

```diff
@@ -767,41 +732,47 @@ oe_result_t oe_sgx_delete_enclave(oe_enclave_t* enclave)
     if (!enclave)
         OE_RAISE(OE_INVALID_PARAMETER);
 
+    /* free allocate memory. */
+    result = _sgx_free_enclave_memory(enclave, enclave->size);
+
+done:
+
+    return result;
+}
+
+oe_result_t _sgx_free_enclave_memory(void* enclave, size_t size)
```

Comment:
Your use of this method is entirely inconsistent: `oe_sgx_create_enclave` calls this with a base address, but `oe_sgx_delete_enclave` calls this with an `oe_enclave_t*` and they are not interchangeable. Instead, define:

```c
oe_result_t _sgx_free_enclave_memory(void* addr, size_t size, bool is_simulation)
```